src/utils/resize-watcher.js

```diff
@@ -0,0 +1,112 @@
+//@flow
+import FakeEvent from '../event/fake-event';
+import FakeEventTarget from '../event/fake-event-target';
+import {CustomEventType} from '../event/event-type';
+
+/**
+ * A Factory class to create a resize observer for the player.
+ */
+class ResizeWatcher extends FakeEventTarget {
+  _observer: ?Object;
+  _playerId: string;
+
+  constructor() {
+    super();
+  }
+
+  /**
+   * Removes resize listeners.
+   * @returns {void}
+   */
+  destroy(): void {
+    if (this._observer) {
+      this._observer.disconnect();
+    }
+    this._observer = null;
+    this._playerId = '';
+  }
+
+  /**
+   * Start listening to a resize of the element.
+   * @param {string} playerId - the target to listen to.
+   * @returns {void}
+   */
+  init(playerId: string): void {
+    this._playerId = playerId;
+    window.ResizeObserver ? this._createNativeObserver() : this._createIframeObserver();
+    const el = document.getElementById(this._playerId);
```

Comment:
Use utlis `getElementById` method
Maybe we need to pass the object already from player to library?

Reply:
yes i think thats better - it would be aligned with native / iframe observer api.